### PR TITLE
Yield more often in tests

### PR DESCRIPTION
**Pull request by findepi**
Some operators may handle yield signal incorrectly. It occurred to me
during my work on #11744 that WIP code did not handle yield, but it was
not visible unless the tests were executing slowly.

By using much shorter processing yield duration (0.001s
instead of 1s), we should have better test coverage for yielding.

### WIP

Note: this is WIP. I have not verified which tests will pass and which will not.

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/execution/executor/TaskExecutor.java [hunk 1]
```diff
@@ -170,17 +173,33 @@ public TaskExecutor(TaskManagerConfig config, EmbedVersion embedVersion, Multile
     @VisibleForTesting
     public TaskExecutor(int runnerThreads, int minDrivers, int guaranteedNumberOfDriversPerTask, int maximumNumberOfDriversPerTask, Ticker ticker)
     {
-        this(runnerThreads, minDrivers, guaranteedNumberOfDriversPerTask, maximumNumberOfDriversPerTask, new EmbedVersion(new ServerConfig()), new MultilevelSplitQueue(2), ticker);
+        this(
+                new Duration(1, SECONDS),
```
Comment (sopel39) [inline presto-main/src/main/java/com/facebook/presto/execution/executor/TaskExecutor.java]: where they are used? Shouldn't they have duration as a parameter too?

File: presto-tests/src/main/java/com/facebook/presto/tests/DistributedQueryRunner.java [hunk 2]
```diff
@@ -186,6 +186,7 @@ private static TestingPrestoServer createTestingPrestoServer(URI discoveryUri, b
                 .put("query.client.timeout", "10m")
                 .put("exchange.http-client.idle-timeout", "1h")
                 .put("task.max-index-memory", "16kB") // causes index joins to fault load
+                .put("task.split-run-duration", "1 us") // causes operators to yield more often
```
Comment (sopel39) [inline presto-tests/src/main/java/com/facebook/presto/tests/DistributedQueryRunner.java]: remove space from `1 us`?

File: presto-main/src/main/java/com/facebook/presto/execution/executor/TaskExecutor.java (hunk 1 above)
Comment (findepi) [reply presto-main/src/main/java/com/facebook/presto/execution/executor/TaskExecutor.java]: `TestTaskExecutor`

File: presto-tests/src/main/java/com/facebook/presto/tests/DistributedQueryRunner.java (hunk 2 above)
Comment (findepi) [reply presto-tests/src/main/java/com/facebook/presto/tests/DistributedQueryRunner.java]: i find this more readable that way

File: presto-tests/src/main/java/com/facebook/presto/tests/DistributedQueryRunner.java [hunk 3]
```diff
@@ -186,6 +186,7 @@ private static TestingPrestoServer createTestingPrestoServer(URI discoveryUri, b
                 .put("query.client.timeout", "10m")
                 .put("exchange.http-client.idle-timeout", "1h")
                 .put("task.max-index-memory", "16kB") // causes index joins to fault load
+                .put("task.split-run-duration", "1 us") // causes operators to yield more often TODO tests should pass with this set to "1 ns" too
```
Comment (sopel39) [inline presto-tests/src/main/java/com/facebook/presto/tests/DistributedQueryRunner.java]: Does that increase total tests runtime?

**Comment by findepi** [comment]
> Have you tested empirically that yield signal is indeed set even before first getOutput call?

There is no guarantee for that. Further, i have 1 µs now.
We could have tests running with yield being set immediately (something like setting this signal without delay, synchronously), but this shouldn't be all our tests. This could hide bugs where operator is misbehaving when yield is not set.

I tested empirically, on my laptop, that this change made my broken implementation of #11744 fail "quite deterministically" (with >80% failure rate on certain test methods, not sure if it was 100%).

**Comment by findepi** [comment]
> Does that increase total tests runtime?

https://travis-ci.org/prestodb/presto/builds/445185387 build was ~45minutes (10%) slower than avg of last 5 builds on master.

**Pull request closed without merging**